core/src/main/java/org/apache/druid/math/expr/vector/VectorProcessors.java

```diff
@@ -135,6 +178,660 @@ public void processIndex(String[] strings, long[] longs, boolean[] outputNulls,
+              String[] rightInput,
+              @Nullable boolean[] rightNulls,
+              int i
+          )
+          {
+            output[i] = leftInput[i] != null ? leftInput[i] : rightInput[i];
+          }
+
+          @Override
+          public ExprEvalVector<String[]> asEval()
+          {
+            return new ExprEvalStringVector(output);
+          }
+        }
+    );
+  }
+
+  public static <T> ExprVectorProcessor<T> not(Expr.VectorInputBindingInspector inspector, Expr expr)
+  {
+    final ExprType inputType = expr.getOutputType(inspector);
+    final int maxVectorSize = inspector.getMaxVectorSize();
+    ExprVectorProcessor<?> processor = null;
+    if (ExprType.STRING.equals(inputType)) {
+      processor = new LongOutStringInFunctionVectorProcessor(expr.buildVectorized(inspector), maxVectorSize)
+      {
+        @Override
+        public void processIndex(String[] strings, long[] longs, boolean[] outputNulls, int i)
+        {
+          outputNulls[i] = strings[i] == null;
+          if (!outputNulls[i]) {
+            longs[i] = Evals.asLong(!Evals.asBoolean(strings[i]));
+          }
+        }
+      };
+    } else if (ExprType.LONG.equals(inputType)) {
+      processor = new LongOutLongInFunctionVectorValueProcessor(expr.buildVectorized(inspector), maxVectorSize)
+      {
+        @Override
+        public long apply(long input)
+        {
+          return Evals.asLong(!Evals.asBoolean(input));
+        }
+      };
+    } else if (ExprType.DOUBLE.equals(inputType)) {
+      processor = new DoubleOutDoubleInFunctionVectorValueProcessor(expr.buildVectorized(inspector), maxVectorSize)
+      {
+        @Override
+        public double apply(double input)
+        {
+          return Evals.asDouble(!Evals.asBoolean(input));
+        }
+      };
+    }
+    if (processor == null) {
+      throw Exprs.cannotVectorize();
+    }
+    return (ExprVectorProcessor<T>) processor;
+  }
+
+  public static <T> ExprVectorProcessor<T> or(Expr.VectorInputBindingInspector inspector, Expr left, Expr right)
+  {
+    final int maxVectorSize = inspector.getMaxVectorSize();
+    return makeSymmetricalProcessor(
+        inspector,
+        left,
+        right,
+        () -> new SymmetricalBivariateFunctionVectorProcessor<long[]>(
+            ExprType.LONG,
+            left.buildVectorized(inspector),
+            right.buildVectorized(inspector)
+        )
+        {
+          final long[] output = new long[maxVectorSize];
+          final boolean[] outputNulls = new boolean[maxVectorSize];
+
+          @Override
+          public void processIndex(
+              long[] leftInput,
+              @Nullable boolean[] leftNulls,
+              long[] rightInput,
+              @Nullable boolean[] rightNulls,
+              int i
+          )
+          {
+            if (NullHandling.sqlCompatible()) {
+              // true/null, null/true, null/null -> true
+              // false/null, null/false -> null
```

Comment:
It seems like this comment doesn't match the implementation

`null /null -> null` which is what's described in `docs/misc/math-expr.md` but this comment says it should be true.